workflow/controller/operator.go

```diff
@@ -2624,3 +2646,60 @@ func (woc *wfOperationCtx) includeScriptOutput(nodeName, boundaryID string) (boo
 	}
 	return false, nil
 }
+
+func (woc *wfOperationCtx) getTopLevelWorkflowTemplate() (*wfv1.WorkflowSpec, error) {
+	var wftmpl wfv1.WorkflowSpecHolder
+	var err error
+	// Logic for Top level Workflow template reference
+	if woc.wf.Spec.WorkflowTemplateRef != nil {
+		if woc.wf.Spec.WorkflowTemplateRef.ClusterScope {
+			wftmpl, err = woc.controller.cwftmplInformer.Lister().Get(woc.wf.Spec.WorkflowTemplateRef.Name)
+		} else {
+			wftmpl, err = woc.controller.wftmplInformer.Lister().WorkflowTemplates(woc.wf.Namespace).Get(woc.wf.Spec.WorkflowTemplateRef.Name)
+
+		}
+		if err != nil {
+			return nil, err
+		}
+	}
+	return wftmpl.GetSpec().DeepCopy(), nil
+}
+
+func (woc *wfOperationCtx) convertWFTmplRefToTmplRef() *wfv1.TemplateRef {
```

Comment:
This function is only used once, so I would just add the code in-place where it is used.